pyteal/compiler/compiler_test.py

```diff
@@ -2193,3 +2193,1141 @@ def access_b4_store(magic_num: pt.abi.Uint64, *, output: pt.abi.Uint64):
 
     with pytest.raises(pt.TealInternalError):
         pt.compileTeal(program_access_b4_store_broken, pt.Mode.Application, version=6)
+
+
+def test_router_app():
+    def add_methods_to_router(router: pt.Router):
+        @pt.ABIReturnSubroutine
+        def add(
+            a: pt.abi.Uint64, b: pt.abi.Uint64, *, output: pt.abi.Uint64
+        ) -> pt.Expr:
+            return output.set(a.get() + b.get())
+
+        router.add_method_handler(add)
+
+        @pt.ABIReturnSubroutine
+        def sub(
+            a: pt.abi.Uint64, b: pt.abi.Uint64, *, output: pt.abi.Uint64
+        ) -> pt.Expr:
+            return output.set(a.get() - b.get())
+
+        router.add_method_handler(sub)
+
+        @pt.ABIReturnSubroutine
+        def mul(
+            a: pt.abi.Uint64, b: pt.abi.Uint64, *, output: pt.abi.Uint64
+        ) -> pt.Expr:
+            return output.set(a.get() * b.get())
+
+        router.add_method_handler(mul)
+
+        @pt.ABIReturnSubroutine
+        def div(
+            a: pt.abi.Uint64, b: pt.abi.Uint64, *, output: pt.abi.Uint64
+        ) -> pt.Expr:
+            return output.set(a.get() / b.get())
+
+        router.add_method_handler(div)
+
+        @pt.ABIReturnSubroutine
+        def mod(
+            a: pt.abi.Uint64, b: pt.abi.Uint64, *, output: pt.abi.Uint64
+        ) -> pt.Expr:
+            return output.set(a.get() % b.get())
+
+        router.add_method_handler(mod)
+
+        @pt.ABIReturnSubroutine
+        def all_laid_to_args(
+            _a: pt.abi.Uint64,
+            _b: pt.abi.Uint64,
+            _c: pt.abi.Uint64,
+            _d: pt.abi.Uint64,
+            _e: pt.abi.Uint64,
+            _f: pt.abi.Uint64,
+            _g: pt.abi.Uint64,
+            _h: pt.abi.Uint64,
+            _i: pt.abi.Uint64,
+            _j: pt.abi.Uint64,
+            _k: pt.abi.Uint64,
+            _l: pt.abi.Uint64,
+            _m: pt.abi.Uint64,
+            _n: pt.abi.Uint64,
+            _o: pt.abi.Uint64,
+            _p: pt.abi.Uint64,
+            *,
+            output: pt.abi.Uint64,
+        ):
+            return output.set(
+                _a.get()
+                + _b.get()
+                + _c.get()
+                + _d.get()
+                + _e.get()
+                + _f.get()
+                + _g.get()
+                + _h.get()
+                + _i.get()
+                + _j.get()
+                + _k.get()
+                + _l.get()
+                + _m.get()
+                + _n.get()
+                + _o.get()
+                + _p.get()
+            )
+
+        router.add_method_handler(all_laid_to_args)
+
+        @pt.ABIReturnSubroutine
+        def empty_return_subroutine() -> pt.Expr:
+            return pt.Log(pt.Bytes("appear in both approval and clear state"))
```

Comment:
note: this doesn't appear in both approval and clear state programs. `log_1` appears in both programs

Reply:
oh boy I typoed it to be `close_out` not `clear_state`, fixing it